src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int128Converter.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a type converter to convert 128-bit signed integer objects to and
+    /// from various other representations.
+    /// </summary>
+    public class Int128Converter : BaseNumberConverter
+    {
+        /// <summary>
+        /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
+        /// </summary>
+        internal override Type TargetType => typeof(Int128);
+
+        /// <summary>
+        /// Convert the given value string to Int128 using the given radix
+        /// </summary>
+        internal override object FromString(string value, int radix)
+        {
+            // We only support decimal and hex radix in the converter.
+            if (radix != 10 && radix != 16)
+            {
+                throw new ArgumentException(SR.Arg_InvalidBase);
```

Comment:
This is an `internal` method that is only ever called with `16`. I don't believe this needs to be an exception or have a resource string.

Reply:
Maybe we can just `Debug.Assert((radix == 10) || (radix == 16))` to catch if we ever change this in the future (such as to support binary, octal, etc)

Reply:
done